src/System.Private.CoreLib/shared/System/Diagnostics/Debug.cs

```diff
@@ -6,15 +6,42 @@
 #define DEBUG
 using System.Diagnostics.Contracts;
 using System.Runtime.CompilerServices;
+using System.Threading;
 
 namespace System.Diagnostics
 {
+    public abstract class DebugProvider
+    {
+        public virtual bool AssertUIEnabled { get { return false; } set { } }
```

Comment:
I do not there needs to be `AssertUIEnabled` here. The checks for AssertUIEnabled can be in the TraceListener override only.

Reply:
Oh right, that would make much more sense. Thanks.